open-api/rest-catalog-open-api.yaml

```diff
@@ -2838,6 +3093,59 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - select
+        - filter
+        - case-sensitive
+      properties:
+        select:
+          description: A list of the selected column names
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Indicates whether column selection and filtering should be case sensitive
+          type: boolean
+        snapshot-id:
+          description: an int64 snapshot ID (if snapshot-range is not present); optional and defaults to the table's current snapshot
+          type: integer
+          format: int64
+        snapshot-range:
+          description: a JSON list containing exactly 2 int64 snapshot IDs (if snapshot-id is not present) representing the start (exclusive) and end (inclusive) snapshots
```

Comment:
Sorry I did not discuss this in the original thread since I was waiting for the merge of content file PR. 

Isn't a range typically inclusive start and exclusive end? why it is reversed here? I am starting to feel maybe using `start-snapshot-id` and `end-snapshot-id` might be a better choice to be less ambiguous, and those values can also be null to indicate no start or no end.

Reply:
> Isn't a range typically inclusive start and exclusive end? why it is reversed here? I am starting to feel maybe using start-snapshot-id and end-snapshot-id might be a better choice to be less ambiguous, and those values can also be null to indicate no start or no end.

+1 for this I think @stevenzwu had also brought this up for the `IncrementalScan` as well having inclusive and exclusive for the from/start snapshot. https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/IncrementalScan.java#L34

Reply:
btw just to clarify my comment, I definitely agree exclusive start and inclusive end is the most common use case in incremental pipelines. I just think it might confuse user because the common "range" in programming language has been inclusive start exclusive end, and not everyone would read the description carefully.

Reply:
Added a flag for user to configure the start of the `snapshot-range` to be exclusive or inclusive. Let me know what you think @jackye1995 @rdblue?